test/e2e/cert_manager_deployment_test.go

```diff
@@ -73,21 +73,23 @@ func TestACMECertsIngress(t *testing.T) {
 	config, err := library.GetConfigForTest(t)
 	require.NoError(t, err)
 
-	loader.CreateTestingNS("hello")
-	defer loader.DeleteTestingNS("hello")
-	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"))
-	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"))
-	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "deployment.yaml"))
-	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "deployment.yaml"))
-	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "service.yaml"))
-	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "service.yaml"))
-
-	routeV1Client, err := routev1.NewForConfig(config)
+	ns, err := loader.CreateTestingNS("e2e-acme-ingress-cert")
 	require.NoError(t, err)
-	route, err := routeV1Client.Routes("openshift-console").Get(ctx, "console", metav1.GetOptions{})
+	defer loader.DeleteTestingNS(ns.Name)
+	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"), ns.Name)
+	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"), ns.Name)
+	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "deployment.yaml"), ns.Name)
+	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "deployment.yaml"), ns.Name)
+	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "service.yaml"), ns.Name)
+	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "service.yaml"), ns.Name)
+
+	configClient, err := configv1.NewForConfig(config)
 	require.NoError(t, err)
+	baseDomain, err := library.GetClusterBaseDomain(ctx, configClient)
+	require.NoError(t, err)
+	appsDomain := "apps." + baseDomain
 
-	ingress_host := "hey." + strings.Join(strings.Split(route.Spec.Host, ".")[1:], ".")
+	ingress_host := "eaic." + appsDomain
```

Comment:
To be robust, need it check if the total lengh would be greater than 64 and if yes, make it shorter?

Reply:
Since, this is an e2e test, I don't think if I need to check the length in my own test code. Because, the cert-manager webhook running on the cluster during the test will always verify the length of the dns name and if it exceeds, it will error out the Create and essentially fail the test, so it's already taken care of in the e2e way, I'd say.